src/internal/ajax/getXHRResponse.ts

```diff
@@ -1,37 +1,11 @@
 /**
- * Gets what should be in the `response` property of the XHR. However,
- * since we still support the final versions of IE, we need to do a little
- * checking here to make sure that we get the right thing back. Consequently,
- * we need to do a JSON.parse() in here, which *could* throw if the response
- * isn't valid JSON.
+ * Gets what should be in the `response` property of the XHR.
  *
  * This is used both in creating an AjaxResponse, and in creating certain errors
  * that we throw, so we can give the user whatever was in the response property.
  *
  * @param xhr The XHR to examine the response of
  */
 export function getXHRResponse(xhr: XMLHttpRequest) {
-  switch (xhr.responseType) {
-    case 'json': {
-      if ('response' in xhr) {
-        return xhr.response;
-      } else {
-        // IE
-        const ieXHR: any = xhr;
-        return JSON.parse(ieXHR.responseText);
-      }
-    }
-    case 'document':
-      return xhr.responseXML;
-    case 'text':
-    default: {
-      if ('response' in xhr) {
-        return xhr.response;
-      } else {
-        // IE
-        const ieXHR: any = xhr;
-        return ieXHR.responseText;
-      }
-    }
-  }
+  return xhr.responseType === 'document' ? xhr.responseXML : xhr.response;
```

Comment:
Honestly, we should just inline this if it's only used in one spot. It's a small amount of code now.

Reply:
[Judging by the specification](https://xhr.spec.whatwg.org/#the-response-attribute), the use of `responseXML` is redundant, since we will get what we need anyway.

> Otherwise, if [this](https://webidl.spec.whatwg.org/#this)’s [response type](https://xhr.spec.whatwg.org/#response-type) is "document", [set a document response](https://xhr.spec.whatwg.org/#document-response) for [this](https://webidl.spec.whatwg.org/#this).

So I replaced `getXHRResponse(xhr)` with `xhr.response`.